src/main/java/org/jabref/model/entry/identifier/DOI.java

```diff
@@ -108,12 +108,12 @@ public DOI(String doi) {
         Matcher matcher = EXACT_DOI_PATT.matcher(trimmedDoi);
         if (matcher.find()) {
             // match only group \1
-            this.doi = matcher.group(1);
+            this.doi = matcher.group(1).replaceAll("[^\\w,/,:,-,.,-]|[_]", "");
```

Comment:
@KunAndrew  I think you need to adjust the DOI regex pattern:
One CrossRef test fails, because of the missing underscore:
  org.opentest4j.AssertionFailedError: expected: <10.1007/11538394_20> but was: <10.1007/1153839420>

Reply:
There should be a comment added which characters are removed and why. Without any comment, this is a "magic number" which should be extraced to a constant.

Reply:
I removed this addtional regex because it was not really making sense. The DOI class already trims DOIs and also removes other encoding components.